src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -93,6 +155,7 @@ protected override void Dispose(bool disposing)
             state.CancellationToken = cancellationToken;
             state.RequestMessage = request;
             state.Handler = this;
+            state.ProxyPolicy = _proxyPolicy;
```

Comment:
Please look at what happens during SendAsync.Am pasting relevant code below for your convenience.
 There am using the value of the _proxyPolicy and not the value of UsePolicy. The value of  UsePolicy is NOT used internally for determining the proxy setting.   


        private static void SetProxyOptions(RequestCompletionSource state, SafeCurlHandle requestHandle)
        {
            var requestUri = state.RequestMessage.RequestUri;
            if (state.ProxyPolicy == CurlProxyUsePolicy.DoNotUseProxy)
            {
                Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_PROXY, string.Empty);
                return;
            }

            if ((state.ProxyPolicy == CurlProxyUsePolicy.UseDefaultProxy) || (state.Proxy == null))
            {
                return;
            }

            Debug.Assert( (state.Proxy != null) && (state.ProxyPolicy == CurlProxyUsePolicy.UseCustomProxy));
            if (state.Proxy.IsBypassed(requestUri))
            {
                Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_PROXY, string.Empty);
                return;
            }
           .... rest of the code to pass the proxy values to curl follows.